color-changer.js

```diff
@@ -0,0 +1,42 @@
+function getRandomUnusedHue(usedHues) {
+    let randomHue;
+    while(randomHue === undefined || usedHues[randomHue]) {
+        randomHue = Math.floor(Math.random()*360);
+    }
+    return randomHue;
+}
+
+function changeBgHue(newHue) {
+    let minLightness = 30;
+    let maxLightness = 70;
+    const randomLightness = Math.floor(Math.random() * (maxLightness - minLightness) + minLightness)
+    document.body.style.backgroundColor = `hsl(${newHue}, 50%, ${randomLightness}%)`;
+}
+
+function changeBgColorRandomly() {
+    const usedHues = {};
+      
+    let randomBgColorChange = function() {
+        let newHue = getRandomUnusedHue(usedHues);
+        changeBgHue(newHue)
+        usedHues[newHue] = true;
+    }
+
+    let count=1;
+    randomBgColorChange();
+    count++;
```

Comment:
Alright, ended up extracting the setInterval part into a function with its own counter and limit. It's a bit more readable than putting the setInterval workaround in changeBgColorRandomly and makes changeBgColorRandomly very clean to read.